### DISCUSSION TITLE
Audio segmentation of radio news and talk show in Non-English language.

### DISCUSSION THREAD

**Discussion by csikasote** [pyannote/pyannote-audio#1149] [Q&A]
Hello, I have a set of audio recordings of radio news and talk show in an African language that I want to automatically segment. The audio durations range from 15 minutes to about 1hr each. Can I use pyannote to automatically segment them? I would to use the segments for unsupervised speech recognition experiments.

Any help will be appreciated.

**Comment by hbredin** [comment]
What did you try? 

I'd recommend you start by using [pyannote/segmentation](https://huggingface.co/pyannote/segmentation) for [voice activity detection](https://huggingface.co/pyannote/segmentation#voice-activity-detection).

**Reply by csikasote** [reply]
Thank you for the recommendation. I was able to run the segmentation for VAD successfully. I have, however, two questions:

1. How do I generate 'segmentation' file that contains all the start and end of each segment.
2. How do I go about making sure that the segments are not more than 20 seconds?

I`m using pyannote for the first time.

**Reply by hbredin** [reply]
1. pyannote.audio pipelines return [pyannote.core.Annotation](http://pyannote.github.io/pyannote-core/structure.html#annotation) instances that can be iterated over to get the segment start time and end time. See [TL;DR section](https://github.com/pyannote/pyannote-audio#tldr-).
2. You would have to postprocess the segments yourself.
